contracts/Graph.sol

```diff
@@ -120,13 +120,15 @@ contract Graph {
         require(withdrawStatus == State.OPENED, "Withdraw is not Opened");
         require(monthlyBudget > _amount, "Insufficent for Withdrawal");
         require(mywithdrawstatus[msg.sender], "Exceeded you withdraw limit");
+
         /** @dev 
          * Calculate the expected Withdraw of a user or organisation
         */
         uint256 expectedWithdraw = uint256(monthlyBudget).mul(uint256(getPageRank(msg.sender))).div(uint256(getTotalPageRank(msg.sender)));
        
         require(expectedWithdraw > monthlyWithdraw[msg.sender]);
-                
+        require(expectedWithdraw > _amount);
```

Comment:
This doesn't consider prior withdrawals, I think it should be 
`require((_amount + monthlyWithdraw[msg.sender]) < allowance)`

Reply:
Please can you explain better

Reply:
Each month, a user is entitled to
`allowance = min(personalBudget, entrustedFunds)`
they may make multiple withdrawals each month, but they must not take more in total than their allowance:
`require((_amount + monthlyWithdraw[msg.sender]) < allowance)`

i.e. I have an allowance of £50, I want to withdraw £40 now, but I've already withdrawn £30 pounds this month

require((40 + 30) < 50) prevents the transaction.

Reply:
if it bases on what the user will take for the month, I think this condition has done         
require(expectedWithdraw > monthlyWithdraw[msg.sender]);
How do you see it

Reply:
No because monthlyWithdraw is not calculated in advance, it only stores how much a person has spent in the past month

Reply:
Good day, please review again